### PR TITLE
anytone-emu: init at unstable-2023-06-15

### PR DESCRIPTION
**Pull request by Janik-Haag**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/anytone-emu/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    udev
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
```
Comment (natsukium) [inline pkgs/applications/radio/anytone-emu/default.nix:36]:
```suggestion
```
I think this application only works on Linux.
It calls a Linux-specific function internally.
https://docs.rs/nix/latest/nix/pty/fn.ptsname_r.html

File: pkgs/applications/radio/anytone-emu/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  meta = with lib; {
+    description = "A tiny emulator for AnyTone radios";
+    homepage = "https://github.com/hmatuschek/anytone-emu";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ janik ];
```
Comment (natsukium) [inline pkgs/applications/radio/anytone-emu/default.nix:43]:
```suggestion
    maintainers = with maintainers; [ janik ];
    platforms = platforms.linux;
```

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 237890` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>anytone-emu</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM.

Since the upstream should maintain cargo.lock in VCS, it might be a good idea to submit an issue.
see, https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html#cargotoml-vs-cargolock

**Comment by Janik-Haag** [comment]
I'll ask upstream to do so, but this shouldn't be a blocker for this pr, right?

**Comment by natsukium** [comment]
Of course, I'll merge it when CI goes green.

**Pull request merged**